### ISSUE TITLE
Style/TernaryParentheses still not necessarily safe autocorrect for `defined?` keyword

### ISSUE THREAD

**Issue by OmriSama** [rubocop-hq/rubocop#8782]
I have a project with this line in it:

```ruby
val = (defined? inc.name) ? inc.name : inc
```

`rubocop -a` autocorrects this to:

```ruby
val = defined? inc.name || inc
```

Which is clearly not the same thing because one returns a Boolean and another returns the `name`.

The correct way to fix this, IMO, to make Rubocop happy:

```ruby
val = defined?(inc.name) ? inc.name : inc
```

I thought this issue was common enough that I would've been able to find someone who had it before, and I did find:

https://github.com/rubocop-hq/rubocop/pull/3505
https://github.com/rubocop-hq/rubocop/issues/3450

But it seems like this problem has persisted...

Thoughts?

## RuboCop version

```
$ bundle exec rubocop -V
0.91.1 (using Parser 2.7.1.4, rubocop-ast 0.4.2, running on ruby 2.7.1 x86_64-darwin19)
```

**Comment by marcandre** [comment]
Thanks for the bug report.

I presume there's another cop involved in the final auto-correction; by itself it corrects to `val = defined? inc.name ? inc.name : inc`, which is incorrect.
